libraries/AP_Math/location.cpp

```diff
@@ -46,6 +46,30 @@ float longitude_scale(const struct Location &loc)
 #endif
 }
 
+// Added for use by SoaringController
+// return distance to north
+float get_offset_north(const struct Location &loc1, const struct Location &loc2)
```

Comment:
This pair of functions are unsafe/poorly specified on return values, as -1.0 is a valid and achievable result, additionally you never actually appear to check the return of -1. (Flying over lat/lng of 0.0 is also valid).